File: java/src/main/java/edu/cmu/cs/cs214/lab02/shapes/Square.java
package edu.cmu.cs.cs214.lab02.shapes;

public class Square implements Shape {
    private final double sideLen;

    public Square(double sideLen) {
        this.sideLen = sideLen;
    }

    public double getArea() {
        return this.getWidth() * this.getHeight(); // should I call function like this?
    }

    public double getWidth() {
        return this.sideLen;
    }

    public double getHeight() {
        return this.sideLen;
    }
}

File: typescript/src/shapes/square.ts
function square(sideLen: number): Shape {
    return {
        getWidth: function (): number {
            return sideLen
        },
        getHeight: function (): number {
            return sideLen
        },
        computeArea: function (): number {
            return sideLen * sideLen
        }
    }
}

export { square }